apps/web/src/routes/AuthLayoutRouter/components/ContextMenu/ContextMenu.jsx

```diff
@@ -84,6 +90,214 @@ export default function ContextMenu (props) {
   const [activeWidget, setActiveWidget] = useState(null)
   const toggleNavMenuAction = useCallback(() => dispatch(toggleNavMenu()), [])
 
+  // Track the newly added top-level widget
+  const newWidget = previousWidgets.length > 0 ? orderedWidgets.find(widget => previousWidgets.indexOf(widget.id) === -1) : null
+  previousWidgets = orderedWidgets.map(widget => widget.id)
+  const newWidgetRef = useRef()
+
+  // Enhanced detection using multiple methods
+  const [newViewId, setNewViewId] = useState(null)
+
+  // Check URL param and localStorage for widget ID
+  useEffect(() => {
+    const newWidgetIdFromUrl = searchParams.get('newWidgetId')
+
+    if (newWidgetIdFromUrl) {
+      console.log('Found new widget ID in URL:', newWidgetIdFromUrl)
+      setNewViewId(newWidgetIdFromUrl)
+
+      // Remove from URL once detected
+      const newParams = new URLSearchParams(searchParams)
+      newParams.delete('newWidgetId')
+      navigate(`${location.pathname}?${newParams.toString()}`, { replace: true })
+    } else {
+      // Check localStorage as backup
+      const storedWidgetId = localStorage.getItem('hylo:last-created-widget')
+      const timestamp = parseInt(localStorage.getItem('hylo:last-created-timestamp') || '0')
+
+      // Only use stored ID if it's recent (within last 5 seconds)
+      if (storedWidgetId && Date.now() - timestamp < 5000) {
+        console.log('Found new widget ID in localStorage:', storedWidgetId)
+        setNewViewId(storedWidgetId)
+        localStorage.removeItem('hylo:last-created-widget')
+        localStorage.removeItem('hylo:last-created-timestamp')
+      }
+    }
+  }, [searchParams, location, navigate])
+
+  // Listen for custom events
```

Comment:
still confused why we are using 3 different methods for this, 1 should be good enough